puppetboard/app.py

```diff
@@ -399,205 +408,122 @@ def node(env, node_name):
-    status_arg = request.args.get('status')
-    reports_query = AndOperator()
-    total_query = ExtractOperator()
-
-    total_query.add_field(FunctionOperator("count"))
-
-    if env != '*':
-        reports_query.add(EqualsOperator("environment", env))
-
-    if status_arg in ['failed', 'changed', 'unchanged']:
-        reports_query.add(EqualsOperator('status', status_arg))
-        reports_query.add(EqualsOperator('noop', False))
-    elif status_arg == 'noop':
-        reports_query.add(EqualsOperator('noop', True))
-
-    if len(reports_query.operations) == 0:
-        reports_query = None
-
-    if reports_query is not None:
-        total_query.add_query(reports_query)
-
-    try:
-        paging_args = {'limit': int(limit)}
-        paging_args['offset'] = int((page - 1) * paging_args['limit'])
-    except ValueError:
-        paging_args = {}
-
-    reports = get_or_abort(puppetdb.reports,
-                           query=reports_query,
-                           order_by=DEFAULT_ORDER_BY,
-                           **paging_args)
-    total = get_or_abort(puppetdb._query,
-                         'reports',
-                         query=total_query)
-    total = total[0]['count']
-    reports, reports_events = tee(reports)
-    report_event_counts = {}
-
-    if total == 0 and page != 1:
-        abort(404)
-
-    for report in reports_events:
-        report_event_counts[report.hash_] = {}
-
-        for event in report.events():
-            if event.status == 'success':
-                try:
-                    report_event_counts[report.hash_]['successes'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['successes'] = 1
-            elif event.status == 'failure':
-                try:
-                    report_event_counts[report.hash_]['failures'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['failures'] = 1
-            elif event.status == 'noop':
-                try:
-                    report_event_counts[report.hash_]['noops'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['noops'] = 1
-            elif event.status == 'skipped':
-                try:
-                    report_event_counts[report.hash_]['skips'] += 1
-                except KeyError:
-                    report_event_counts[report.hash_]['skips'] = 1
     return Response(stream_with_context(stream_template(
         'reports.html',
-        reports=yield_or_stop(reports),
-        reports_count=app.config['REPORTS_COUNT'],
-        report_event_counts=report_event_counts,
-        pagination=Pagination(page, paging_args.get('limit', total), total),
         envs=envs,
         current_env=env,
-        limit=paging_args.get('limit', total))))
-
+        table_count=app.config['NORMAL_TABLE_COUNT'],
+        table_count_selector=app.config['TABLE_COUNT_SELECTOR'],
```

Comment:
The templates have direct access to the config settings because it's rendered in the Flask application's context. `app.config['LITTLE_TABLE_COUNT']` and `app.config['TABLE_COUNT_SELECTOR']` are available in the templates as `config.LITTLE_TABLE_COUNT` and `config.TABLE_COUNT_SELECTOR` respectively.